drivers/usb/usbccgp/function.c

```diff
@@ -654,24 +654,61 @@ USBCCGP_LegacyEnum(
             return STATUS_UNSUCCESSFUL;
         }
 
-        //
-        // init function descriptor
-        //
-        FDODeviceExtension->FunctionDescriptor[Index].FunctionNumber = Index;
-        FDODeviceExtension->FunctionDescriptor[Index].NumberOfInterfaces = 1;
-        FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList = AllocateItem(NonPagedPool, sizeof(PUSB_INTERFACE_DESCRIPTOR) * 1);
-        if (!FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList)
+        SubIndex = 0;
+        if (InterfaceDescriptor->bInterfaceClass == USB_DEVICE_CLASS_AUDIO)
         {
+            // AUDIO CLASS lets group all audio interfaces together
+            //
+            // init function descriptor
+            //
+            FDODeviceExtension->FunctionDescriptor[Index].FunctionNumber = Index;
+            FDODeviceExtension->FunctionDescriptor[Index].NumberOfInterfaces = 1;
+            FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList = AllocateItem(NonPagedPool, sizeof(PUSB_INTERFACE_DESCRIPTOR) * FDODeviceExtension->ConfigurationDescriptor->bNumInterfaces);
+            if (!FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList)
+            {
+                //
+                // no memory
+                //
+                return STATUS_INSUFFICIENT_RESOURCES;
+            }
+
             //
-            // no memory
+            // store interface descriptor
             //
-            return STATUS_INSUFFICIENT_RESOURCES;
+            FDODeviceExtension->FunctionDescriptor[Index].InterfaceDescriptorList[SubIndex] = InterfaceDescriptor;
+            do
```

Comment:
`do { ... } while (TRUE);` --> `while (TRUE) { ... }` , or: `for (;;) { ... }` instead.
This allows to see it's an infinite loop (a priori) without having to scroll all the way down to the trailing "while".